### DISCUSSION TITLE
Snapcast - Better Handling of Multiroom through custom Stream Name?

### DISCUSSION THREAD

**Discussion by almostserious** [music-assistant/support#2578] [Q&A]
Hi everyone,

I have a question or an idea about improving the handling of Snapcast players. 
This might not be the optimal approach for everyone, but it’s how I personally use Snapcast. 
If there's already a smart way to achieve this within MA, I'd love to learn more.
Maybe im thinking too complicated.

**Current Setup:**
I have several Raspberry Pi (RPi) Snapclients and some Android clients running.
I've created groups per room with different clients.

**The Issue:**
When using Music Assistant (MA) and Snapcast, I have to either:
Add an extra group with all players, or
Manually add the clients to an existing stream.
The MA UI for adding or playing on specific clients is super confusing to me.
No matter how often I use it, i tend to make mistakes and it never plays where I want in the end.

So far, I haven't found a way to do this through the Home Assistant (HA) UI as well. 

**My Main Use Case:**
I play music centrally and dynamically add or remove clients. 
Previously, before using MA, I would play to my Snapserver instance and then mute/unmute clients as needed.

However, with MA, I haven't found a way to replicate this behavior.
Whenever I stream to one or more clients, MA creates a new stream with a random name. 
To automate this via the HA UI, the stream name should be consistent.
Then I could simply call an automation that changes the stream name of my groups to the correct one. 

**Question:**
Is it possible to set a custom standard stream name for MA?

**Comment by marcelveldt** [comment]
What does the stream name have to do with that? I think you should either use MA as intended or stick with Snapcast as standalone situation.

So just use the MA services to join players, not Snapcast

**Comment by almostserious** [comment]
Ok, maybe the Problems stem from the fact that the MA services don't seem to work for me with Snapcast at all.
I just tried it again. 

Several Issues:
1. When I play to a snapcast group that i created within MA the music only plays on a subset of speakers and not all members of this group.
However, in MA it shows as all members playing while Snapcast shows that only 2 members changed to the new stream name.
2. After starting to play, im unable to interact with the stream anymore. Pausing, Playing, Forward etc. all not possible. I get error message NONE in MA.

Only way to stop music is by turning off the Group again. 
Maybe worth to open an Issue and to post Log Files.


One more question, i couldn't find a service call that adds a player dynamically to an already playing group. 
Is this also done via the normal mass.play_media ?

**Reply by OzGav** [reply]
> i couldn't find a service call that adds a player dynamically to an already playing group.

`media_player.join` should work https://www.home-assistant.io/integrations/media_player/
